### ISSUE TITLE
Value object Java serialization definition improvements

### ISSUE THREAD

**Issue by jbergfi** [immutables/immutables#114]
Right now defining java serialization support is verbose. This is unnecessary when you have a lot of value objects that all should be java serializable. Immutables could generate the java serialization definition based on annotations.

For example this:

```java
@Immutable
static abstract class AbstractValue implements Serializable {

	private static final long serialVersionUID = 1L;
	
	public abstract int foo();
	public abstract String bar();
}
```

could be reduced to this:

```java
@Immutable
static abstract class AbstractValue {
	public abstract int foo();
	public abstract String bar();
}
```

And the definition could be for example @JavaSerialize(1L) which could be set on either package or class.

**Comment by elucash** [comment]
Currently implemented serialization approach is basic/minimal and what is important, requires no API elements provided by _Immutables_: just using `java.io.Serializable` or declaring `private static final long serialVersionUID = 1L` will generate all the supporting code.

#113 Will expand the API, probably, having separate API module/jar. The API could then contain annotations like `@Serial.Structural`, `@Serial.Simple` `@Serial.Version`.

This issue is to ensure that new annotation API created in #113 could be applied to umbrella types and packages, or even serve as meta-annotation akin to style annotations.

Thanks for feature request!

**Comment by elucash** [comment]
Implemented as part of #113, released in 2.0.12
